### PR TITLE
division and subtraction have been implemented and tested. This shoul…

### PR DESCRIPTION
**Pull request by William-Charles**
…d have been multiple commits.

### CODE REVIEW

File: src/CellularParser.test.js [hunk 1]
```diff
@@ -22,36 +20,70 @@ shouldParseAs.only = (input, expected) => {
...
+      TokenTypes.TIMES,
+      new NumberExpression("2")
+    ),
+    TokenTypes.MINUS,
+    new NumberExpression("3")
+  )
+);
+
+shouldParseAs(
+  "2 / 1",
```
Comment (lancefisher) [inline src/CellularParser.test.js]: This is good. I'd like to add at least one test that verifies precedence with `/`.

File: src/Cell.js [hunk 2]
```diff
@@ -25,8 +25,8 @@ export default class Cell {
   }
 
   isLiteral() {
-    if (typeof this.text === 'number') return true;
-    if (this.text[0] !== '=') return true;
+    if (typeof this.text === "number") return true;
+    if (this.text[0] !== "=") return true;
```
Comment (lancefisher) [inline src/Cell.js]: I've standardized on single quotes. Can you leave them, please?

**Review by lancefisher — CHANGES_REQUESTED** [review]
Hi Will, thanks for the PR! This looks good. Just need to add a test and use single quotes.

**Comment by William-Charles** [comment]
Made the changes requested and I am getting all green tests.

**Comment by William-Charles** [comment]
I made the changed the double quotes back to single and added a division
precedence test to the interpreter.test file. The test is dividing before
addition.

File: src/Cell.sublime-workspace [hunk 3]
```diff
@@ -0,0 +1,574 @@
+{
```
Comment (lancefisher) [inline src/Cell.sublime-workspace]: Please exclude this file. You can add `src/Cell.sublime-workspace` to `.gitignore`.

File: src/tokenizer.test.js [hunk 4]
```diff
@@ -33,18 +29,18 @@ tokensShouldBe('SUM(A1:B2)', [
   new Token(TokenTypes.COLON, ':'),
   new Token(TokenTypes.NAME, 'B2'),
   new Token(TokenTypes.RPAREN, ')'),
-  new Token(TokenTypes.EOF),
+  new Token(TokenTypes.EOF)
```
Comment (lancefisher) [inline src/tokenizer.test.js]: Looks like a bunch of these changes are unrelated. I think your editor removed the trailing commas here. I want to leave them to follow the [airbnb styleguide](https://github.com/airbnb/javascript).

File: src/tokenizer.test.js [hunk 5]
```diff
@@ -99,19 +95,33 @@ tokensShouldBe('COUNTIF(D5:D11,"red")', [
...
   new Token(TokenTypes.NUMBER, '2'),
-  new Token(TokenTypes.EOF),
+  new Token(TokenTypes.EOF)
+]);
+
+tokensShouldBe('2 - 1', [
+  new Token(TokenTypes.NUMBER, '2'),
+  new Token(TokenTypes.MINUS, '-'),
+  new Token(TokenTypes.NUMBER, '1'),
+  new Token(TokenTypes.EOF)
```
Comment (lancefisher) [inline src/tokenizer.test.js]: Please run `yarn lint` and fix linting errors. Trailing commas are a lint rule.

File: src/tokenizer.js [hunk 6]
```diff
@@ -49,24 +49,34 @@ function tokenize(input) {
...
     }
+    if (char === '/') {
+      tokens.push(new Token(TokenTypes.DIVIDE, '/'));
+      current += 1;
+      continue;
+    }
 
     if (char === '"') {
-      let value = '';
+      let value = "";
```
Comment (lancefisher) [inline src/tokenizer.js]: This should be `''` (single quotes).

File: src/tokenizer.js [hunk 7]
```diff
@@ -89,9 +99,10 @@ function tokenize(input) {
     if (NUM.test(char)) {
       let value = '';
       let foundDot = false;
-      while (char !== undefined && (NUM_OR_DOT.test(char))) {
+      while (char !== undefined && NUM_OR_DOT.test(char)) {
         const isDot = char === '.';
-        if (foundDot && isDot) throw new TypeError('Invalid number format x.x.');
+        if (foundDot && isDot)
+          throw new TypeError('Invalid number format x.x.');
```
Comment (lancefisher) [inline src/tokenizer.js]: No need to change this to two lines. Prefer to keep simple checks on one line.

File: src/tokenizer.js [hunk 8]
```diff
@@ -102,16 +113,12 @@ function tokenize(input) {
...
   tokens.push(new Token(TokenTypes.EOF));
   return tokens;
 }
 
-export {
-    tokenize,
-    Token,
-    TokenTypes,
-};
+export { tokenize, Token, TokenTypes };
```
Comment (lancefisher) [inline src/tokenizer.js:123]: Not that big of deal in this case, but avoid making style changes like this that aren't related to the addition.

**Review by lancefisher — CHANGES_REQUESTED** [review]

**Comment by William-Charles** [comment]
Hey Lance,
Thank you for being patient with me and allowing me to workout me editor
auto changes.
Will

**Comment by lancefisher** [comment]
No worries! It would be nice to clean up the PR title and description too. Explain what you're adding here, etc. It serves as a historical record for the project. Thanks for the PR!

**Comment by lancefisher** [comment]
Tests and linting passed. Looks good.

**Review by lancefisher — APPROVED** [review]

**Pull request merged**